### PR TITLE
[Stable plugin api] Clean up comments and variable names in gradle

### PR DESCRIPTION
**Pull request by pgomulka**
in example plugins a pluginApiVersion is more appropriate than elasticsearchVersion as it better implies that it can be different then the version of ES cluster

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against main? Unless there is a good reason otherwise, we prefer pull requests against main and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

File: build-tools/src/main/resources/plugin-descriptor.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Elasticsearch plugin descriptor file
-# This file must exist as 'plugin-descriptor.properties' inside a plugin.
+# This file must exist as 'plugin-descriptor.properties' or 'stable-plugin-descriptor.properties inside a plugin.
```
Comment (thecoop) [inline build-tools/src/main/resources/plugin-descriptor.properties:2]: Somewhat unrelated - is the name of this file still ok?

**Comment by pgomulka** [comment]
@elasticmachine update branch

File: build-tools/src/main/resources/plugin-descriptor.properties (hunk 1 above)
Comment (pgomulka) [reply build-tools/src/main/resources/plugin-descriptor.properties:2]: good point, perhaps plugin-descriptor-template.properties? or something along these lines (ideas?)
this is a file that is used by both stable and old-style gradle plugins to generate plugin-descriptor.properties (so this file but with values populated)

Comment (thecoop) [reply build-tools/src/main/resources/plugin-descriptor.properties:2]: Yeah, 'template' is a good idea, that would work

**Review by thecoop — APPROVED** [review]

**Pull request merged**